Source/fsSerializer.cs

```diff
@@ -474,9 +477,61 @@ internal class fsLazyCycleDefinitionWriter {
 
             var result = InternalSerialize_1_ProcessCycles(storageType, instance, out data);
             Invoke_OnAfterSerialize(processors, storageType, instance, ref data);
+
+            //remove default values serialization in the end?
+            if (!fsConfig.SerializeDefaultValues){
+                return RemoveDefaultValues(storageType, instance, ref data);
+            }
+
             return result;
         }
 
+        ///Clean up defaults. Is there a better way to do this?
+        private fsResult RemoveDefaultValues(Type storageType, object instance, ref fsData data){
```

Comment:
Can you explain this algorithm? How well does it work?

What about checking inside of TrySerialize (before we start going into Serialize_1, Serialize_2, etc) if instance is equal to its default. If it is, then we return either null or a special 'default' fsData instance that the JSON printer will skip? That way for default values we skip going through a converter (ie, special case the 'default' case like the 'null' case).

It'd also be great if we could use something other than null - maybe

```
class fsData {
    private static object _default = new object();

    private void SetDefault() {
        _instance = _default;
    }
}
```

If this poses an issue because we don't have a type anymore, we could have IsNull also return true for default values.